pkg/server/api/agent/v1/service.go

```diff
@@ -516,23 +514,19 @@ func (s *Service) getSelectorsFromAgentID(ctx context.Context, agentID string) (
 func (s *Service) attestJoinToken(ctx context.Context, token string) (*nodeattestor.AttestResult, error) {
 	log := rpccontext.Logger(ctx).WithField(telemetry.NodeAttestorType, "join_token")
 
-	resp, err := s.ds.FetchJoinToken(ctx, &datastore.FetchJoinTokenRequest{
-		Token: token,
-	})
+	resp, err := s.ds.FetchJoinToken(ctx, token)
```

Comment:
nit: rename `resp` to `joinToken`. If that is too similar to `token`, you can also rename `token` to something like `tokenValue`